internal/frontend/middleware/global.go

```diff
@@ -95,15 +98,21 @@ func Setup(opts *Options) {
 	authBasic = opts.AuthBasic
 	authToken = opts.AuthToken
 	appLogger = opts.Logger
+	basePath = opts.BasePath
 }
 
 func prefixChecker(next http.Handler) http.Handler {
 	return http.HandlerFunc(
 		func(w http.ResponseWriter, r *http.Request) {
-			if strings.HasPrefix(r.URL.Path, "/api") {
-				next.ServeHTTP(w, r)
+			if basePath != "" && r.URL.Path == "/" {
+				http.Redirect(w, r, basePath, http.StatusSeeOther)
+				return
+			}
+
+			if strings.HasPrefix(r.URL.Path, path.Join(basePath, "/api")) {
```

Comment:
Is there a better solution to `path.Join` here? It doesn't feel robust. 

It really feels like `/api` should be a handler which is hung off of `http.StripPrefix(basePath, next)` such that any route registered under it would be handled by `/api`

I wasn't sure of the consequences of  such a change however, so I just updated the string comparison.

Reply:
See https://github.com/dagu-org/dagu/pull/714#discussion_r1845554710 this has been improved to not rely on path.join for this prefix check